src/iminuit/util.py

```diff
@@ -1475,3 +1561,21 @@ def _smart_sampling(f, xmin, xmax, start=5, tol=5e-3):
     xy = list(y.items())
     xy.sort()
     return np.transpose(xy)
+
+
+def _get_annotations(obj: Any) -> Dict[str, Any]:
+    # Compatibility wrapper for CPython < 3.10,
+    # see https://docs.python.org/3/howto/annotations.html
+    #
+    # Returns a dict that maps parameter names to annotations.
+    # If a type is not annotated, it is ommitted.
+    try:
```

Comment:
This should be based on `sys.version_info`, not try/except with type ignores.

Reply:
Why?

Reply:
One of the reasons, but yes. Code is better than comments - notice you put a comment in saying that this is due to Python version. Using `sys.version_info` makes the comment unnecessary, as it's now expressed in code, and static tools (like mypy, as you noted) can use the information. Tools like pyupgrade can detect and remove dead code blocks when you upgrade, to keep this from becoming technical debt. I generally grep for `version_info` when dropping a Python version, as well, just in case pyupgrade misses some.

Reply:
> Code is better than comments 

True.